### PR TITLE
Feature17 select nlp api

### PR DESCRIPTION
**Pull request by haitham-awad**

### CODE REVIEW

**Comment by haitham-awad** [comment]
[Task 17: Research and Select a Free AI NLP API (e.g., Hugging Face, OpenAI)](https://trello.com/c/ZIFKbFjc/21-task-17-research-and-select-a-free-ai-nlp-api-eg-hugging-face-openai)

**Review by jerkur22 — DISMISSED** [review]

File: backend/utils/natural_language_processing_api.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
+from dotenv import load_dotenv
+import json
+from groq import Groq, AuthenticationError, APIConnectionError
+import os
+
+
+def prompt_nlp_model(prompt, api_inputted_key):
```
Comment (arikamat) [inline backend/utils/natural_language_processing_api.py]: lets use .env to get API Key and reformulate the tests

File: backend/utils/natural_language_processing_api.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+from groq import Groq, AuthenticationError, APIConnectionError
+import os
+
+
+def prompt_nlp_model(prompt, api_inputted_key): 
+    """
+    Prompts the API and with information
+
+    Args:
+        str: A question or command for the model to fulfill
```
Comment (arikamat) [inline backend/utils/natural_language_processing_api.py:12]: it takes 2 parameters but only one param is documented. However, if you plan  to get rid of the api_key parameter, this wont be a problem anymore

File: backend/utils/natural_language_processing_api.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+                    "content": prompt,
+                }
+            ],
+            model="llama3-8b-8192",
+
+        )
+
+    #if the API key is invalid
+    except AuthenticationError:
+        return "API KEY IS INVALID"
```
Comment (arikamat) [inline backend/utils/natural_language_processing_api.py]: I wouldn't return a string as the error.
i would return `{"error": "API KEY is invalid"}`

**Review by arikamat — CHANGES_REQUESTED** [review]

File: tests/backend/test_utils_natural_language_processing_api.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+    Makes sure that nothing happens if there is no API key
+    """
+
+    #do not need to do a mock response since not using our API Key
+    response_json = prompt_nlp_model("Hi","")
+    
+
+    #if no API key provided
+    assert response_json == "NO API KEY PROVIDED"
```
Comment (arikamat) [inline tests/backend/test_utils_natural_language_processing_api.py]: will require a modification if you change error behavior

File: tests/backend/test_utils_natural_language_processing_api.py [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    Tests the prompt_npl_model function
+
+    Makes sure that nothing happens if the API key is invalid
+    """
+
+    #do not need to do a mock response since not using our API Key
+    response_json = prompt_nlp_model("Hi","invalid_API_key")
+    
+    #if API key is invalid
+    assert response_json == "API KEY IS INVALID"
```
Comment (arikamat) [inline tests/backend/test_utils_natural_language_processing_api.py]: will require a modification if you change error behavior

**Review by arikamat — APPROVED** [review]

File: tests/backend/test_utils_natural_language_processing_api.py [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+    #do not need to do a mock response since not using our API Key
+    response_json = prompt_nlp_model("Hi")
+    
+    #if API key is invalid
+    assert response_json == {"error": "API KEY is invalid"}
+    
+
+def test_missing_api_key(monkeypatch):
+    """
```
Comment (arikamat) [inline tests/backend/test_utils_natural_language_processing_api.py:47]: documentation incomplete

File: tests/backend/test_utils_natural_language_processing_api.py [hunk 7]
```diff
@@ -0,0 +1,63 @@
+import pytest
+from backend.utils import prompt_nlp_model
+from dotenv import load_dotenv
+import os
+import io
+import sys
+
+def test_valid_prompt_and_api_key():
+    """
```
Comment (arikamat) [inline tests/backend/test_utils_natural_language_processing_api.py:9]: documentation incomplete

**Review by arikamat — CHANGES_REQUESTED** [review]

**Review by arikamat — APPROVED** [review]

**Review by jerkur22 — APPROVED** [review]

**Pull request merged**